app/src/main/java/com/glia/exampleapp/Utils.java

```diff
@@ -17,7 +17,11 @@
 class Utils {
 
     public static String getStringFromPrefs(@StringRes int keyValue, String defaultValue, SharedPreferences sharedPreferences, Resources resources) {
-        return sharedPreferences.getString(resources.getString(keyValue), defaultValue);
+        String value = sharedPreferences.getString(resources.getString(keyValue), defaultValue);
+        if (value.trim().isEmpty()) {
+            return null;
+        }
+        return value;
```

Comment:
I doubt if it won't break the code which already expects that an empty string for default value returns an empty string instead of `null`.
Unless you are 100% sure it won't, I would rather change the validation in `MainFragment` to `if (visitorContext is not empty) `.

Reply:
E/CallVisualizerRepository: Error during accepting engagement request, reasonvisitorContextAssetId must be UUID

Empty value is not null and can cause unexpected error while testing. That is why the change

Reply:
Yes, I understand this. 

And what I suggested is to avoid changing the `getStringFromPrefs()` common function behavior. Instead, change `MainActivity:384` from `if (visitorContext != null)` to `if (visitorContext != null && visitorContext.trim().isEmpty())`.

Does it make sense to you?

Reply:
The main reason for this suggestion is that I am not sure that if you change the behavior of the function (replace empty string with `null`) which is used in many other places, it will not break anything for current usages. Are you sure in this?